### PR TITLE
post: angular cli 1.4

### PR DESCRIPTION
**Pull request by cexbrayat**
This is planned for thursday to give you time to review

### CODE REVIEW

File: _posts/2017-09-14-angular-cli-1.4.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+layout: post
+title: What's new in Angular CLI 1.4?
+author: cexbrayat
+tags: ["Angular 2", "Angular", "Angular 4", "Angular CLI"]
+description: "Angular CLI 1.4 is out! Which new features are included?"
+---
+
+[Angular CLI 1.4.0](https://github.com/angular/angular-cli/releases/tag/v1.4.0) is out with some nice new features!
+
+You can check out what files you need to update using [angular-cli-diff](https://github.com/cexbrayat/angular-cli-diff),
```
Comment (clacote) [inline _posts/2017-09-14-angular-cli-1.4.md]: You should explain where `angular-cli-diff` comes from. Without explanation, it looks like an official tool.

File: _posts/2017-09-14-angular-cli-1.4.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+This is a feature that other frameworks also have (like [Vue CLI](https://github.com/vuejs/vue-cli#official-templates) or [Ember CLI](https://ember-cli.com/extending/)).
+
+We can expect to see new "schematics" for Angular CLI really soon,
+for example some oriented for server-side rendering apps, or mobile apps, or progressive apps, or native apps, etc.
+
+The CLI now has an option to specify the blueprint you want to use when you generate a project:
+
+    ng new --collection my-custom-schematics project-name
+
+You can of course define your own schematics, even we did not give it a try yet.
```
Comment (clacote) [inline _posts/2017-09-14-angular-cli-1.4.md]: `even` looks not right. `but`, `besides`, ... ?

File: _posts/2017-09-14-angular-cli-1.4.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+## Serve path
+
+A new option for `ng serve` is available to specify a path where you want the application to be served in dev:
+
+    ng serve --serve-path hello
+
+will serve the application at `http://localhost:4200/hello`.
+This is a simpler way to configure it, even if you could do the same with the `--base-href` and `--deploy-url` flags.
```
Comment (clacote) [inline _posts/2017-09-14-angular-cli-1.4.md]: I'm not sure that you can use a comparative form ("simpler") without saying upon what it is compared ("simpler... than..."). But you know probably better than me.

File: _posts/2017-09-14-angular-cli-1.4.md (hunk 1 above)
Comment (jnizet) [reply _posts/2017-09-14-angular-cli-1.4.md]: You shouldn't be shy and tell that you created that project. I'd say something like

```
If you want to upgrade to 1.4.0 without pain (or to any other version, BTW), I have created a github project to help: [angular-cli-diff](https://github.com/cexbrayat/angular-cli-diff). Choose the version you're currently using (1.2.1 for example), and the target version (1.4.0 for example), and it gives you a diff of all files created by the CLI: [angular-cli-diff/compare/1.2.1…1.4.0](https://github.com/cexbrayat/angular-cli-diff/compare/1.2.1...1.4.0). You have no excuse for staying behind anymore.

File: _posts/2017-09-14-angular-cli-1.4.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+Angular CLI will now use this new tool to generate the application skeleton
+and the components, services, pipes...
+All the blueprints are now bundled in [`schematics/@angular`](https://github.com/angular/devkit/tree/master/packages/schematics/angular)
+and have been removed from the CLI itself.
+
+Really interesting as it opens the possibility to have other blueprints than the official one!
+This is a feature that other frameworks also have (like [Vue CLI](https://github.com/vuejs/vue-cli#official-templates) or [Ember CLI](https://ember-cli.com/extending/)).
+
+We can expect to see new "schematics" for Angular CLI really soon,
+for example some oriented for server-side rendering apps, or mobile apps, or progressive apps, or native apps, etc.
```
Comment (jnizet) [inline _posts/2017-09-14-angular-cli-1.4.md]: We should create one for resolvers.

File: _posts/2017-09-14-angular-cli-1.4.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+## Schematics
+
+The Google team has created [schematics](https://github.com/angular/devkit/tree/master/packages/angular_devkit/schematics),
+a Yeoman-like generator.
+Angular CLI will now use this new tool to generate the application skeleton
+and the components, services, pipes...
+All the blueprints are now bundled in [`schematics/@angular`](https://github.com/angular/devkit/tree/master/packages/schematics/angular)
+and have been removed from the CLI itself.
+
+Really interesting as it opens the possibility to have other blueprints than the official one!
```
Comment (jnizet) [inline _posts/2017-09-14-angular-cli-1.4.md]: one -> ones

File: _posts/2017-09-14-angular-cli-1.4.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+The CLI now has an option to specify the blueprint you want to use when you generate a project:
+
+    ng new --collection my-custom-schematics project-name
+
+You can of course define your own schematics, even we did not give it a try yet.
+
+## Serve path
+
+A new option for `ng serve` is available to specify a path where you want the application to be served in dev:
```
Comment (jnizet) [inline _posts/2017-09-14-angular-cli-1.4.md]: I guess the same option exists for ng build. Might be worth mentioning it:

    A new option for `ng serve` (and `ng build`)

File: _posts/2017-09-14-angular-cli-1.4.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+This is a simpler way to configure it, even if you could do the same with the `--base-href` and `--deploy-url` flags.
+
+## Missing translation strategy
+
+If you are using the i18n support of Angular,
+you can specify directly from the CLI the strategy you want to adopt when a translation is missing:
+
+    ng build --aot --locale fr --i18n-file src/i18n/messages.fr.xlf --missing-translation error
+
+The `--missing-translation` error flag is available for the `serve` and `build` command,
```
Comment (jnizet) [inline _posts/2017-09-14-angular-cli-1.4.md]: command -> commands

File: _posts/2017-09-14-angular-cli-1.4.md (hunk 1 above)
Comment (cexbrayat) [reply _posts/2017-09-14-angular-cli-1.4.md]: 👍 I also plan to write a proper announcement blog post soon

File: _posts/2017-09-14-angular-cli-1.4.md (hunk 4 above)
Comment (cexbrayat) [reply _posts/2017-09-14-angular-cli-1.4.md]: There is a pending PR https://github.com/angular/angular-cli/pull/6016 I added a comment to ask if that should be reopened on schematics.

File: _posts/2017-09-14-angular-cli-1.4.md (hunk 6 above)
Comment (cexbrayat) [reply _posts/2017-09-14-angular-cli-1.4.md]: I think this is a `serve` only option https://github.com/angular/angular-cli/search?q=servePath&type=Code&utf8=%E2%9C%93

**Comment by cexbrayat** [comment]
Should be ready to merge and tweet tomorrow morning (cc @clacote )

**Pull request merged**